addons/edi_sale_notifier/tests/test_sale_notifier.py

```diff
@@ -0,0 +1,76 @@
+from odoo.addons.edi_sale.tests.common import EdiSaleCase
+from odoo import fields
+
+from unittest.mock import patch
+
+
+class TestSaleNotifier(EdiSaleCase):
+    """EDI sale order request tests"""
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.sale_request_tutorial_record_type = cls.env.ref("edi_sale.sale_request_tutorial_record_type")
+        cls.sale_line_request_tutorial_record_type = cls.env.ref("edi_sale.sale_line_request_tutorial_record_type")
+        cls.doc_type_tutorial = cls.env.ref("edi_sale.sale_request_tutorial_document_type")
+        cls.doc_type_tutorial.active = True
+        cls.sale_request_tutorial_record_type.active = True
+        cls.sale_line_request_tutorial_record_type.active = True
+
+    @classmethod
+    def create_tutorial(cls, *filenames, fail_fast=True):
+        """Create sale order request tutorial document"""
+        cls.doc_type_tutorial.fail_fast = fail_fast
+        return cls.create_input_document(cls.doc_type_tutorial, *filenames)
+
+    def prepare_document(self, doc):
+        """Actions taken when preparing document"""
+        SaleRequestTutorialRecord = self.env["edi.sale.request.tutorial.record"]
+
+        res = doc.action_prepare()
+        # Find related records and fill in client_order_ref and requested_date
+        # (These are fields that are made required in other addons like udes_sale_stock
+        # that are not in the manifest so these need to be populated when running tests
+        # with both addons installed)
+        #
+        # records = SaleRequestTutorialRecord.search([("doc_id", "=", doc.id)])
+        # records.write(
+        #     {
+        #         "client_order_ref": "test_reference",
+        #         "requested_date": fields.Datetime.now()
+        #     }
+        # )
+        return res
```

Comment:
This might be needed to be enabled again but it will make the tests fail if the other module (udes_sale_stock) is not installed.

Reply:
Why does it fail if `udes_sale_stock` is not installed? Can you post the error?

Reply:
Every test fails with: ValueError: Invalid field 'client_order_ref' on model 'edi.sale.request.tutorial.record'
This is because client_order_ref is defined in "udes_sale_stock".

Reply:
@PeterC10 did you have time to check this?

Reply:
So wouldn't this test fail on CI as udes_sale_stock will be installed? And the fix is commented out?

Reply:
Which CI? At the moment this module cannot run here as it relies on edi_notifier which relies on edi.
CI on odoo_edi isn't running as far as I know, I have asked around if this is okay but did not get a reply.

Reply:
I'm a bit confused. `edi_notifier` is in `udes_open`, does the `udes_open` CI not pull the EDI repo atm?

If not then let's make an issue to resolve.

Reply:
I don't think the `udes_open` CI pulls the EDI repo, we'll need an issue to resolve

Reply:
That's correct, `udes_open` does not pull EDI repo.

Reply:
We need to review the Odoo 14 `udes_open` CI Dockerfile, at which point we can resolve the issues with the tests.